cutplace/sql.py

```diff
@@ -52,175 +51,75 @@ def assert_is_valid_dialect(dialect):
-                      str(range_rule.upper_limit) + " )"
-        if i < len(range_rule.items) - 1:
-            constraint += " or "
-        else:
-            constraint += " )"
-
-    return [column_def, constraint]
-
-
-def as_sql_date(field_name, field_is_allowed_to_be_empty, human_readable_format, db):
-    column_def = ""
-    constraint = ""
-
-    if "hh" in human_readable_format and "YY" in human_readable_format:
-        column_def = field_name + " datetime"
-    elif "hh" in human_readable_format:
-        column_def = field_name + " time"
-    else:
-        column_def = field_name + " date"
-
-    if not field_is_allowed_to_be_empty:
-        column_def += " not null"
+def write_create(cid_path, cid_reader):
+    cid_reader.read(cid_path, rowio.excel_rows(cid_path))
 
-    return [column_def, constraint]
+    create_path = os.path.splitext(cid_path)[0] + '_create.sql'
+    # TODO: Add option to specify target folder for SQL files.
+    _log.info('write SQL create statements to "%s"', create_path)
+    with io.open(create_path, 'w', encoding='utf-8') as create_file:
+        # TODO: Add option for encoding.
+        sql_factory = SqlFactory(cid_reader, os.path.splitext(cid_path)[0])
+        create_file.write(sql_factory.create_table_statement())
+        # TODO: Add option for target SQL dialect
 
 
-def as_sql_create_table(cid, dialect='ansi'):
-    assert_is_valid_dialect(dialect)
+class SqlFactory(object):
+    def __init__(self, cid, table, dialect=ANSI):
+        self._cid = cid
+        self._table = table
+        self._dialect = dialect
 
-    file_name = os.path.basename(cid._cid_path)
-    table_name = file_name.split('.')
+        assert_is_valid_dialect(self._dialect)
 
-    result = "create table " + table_name[0] + " (\n"
-    constraints = ""
+    @property
+    def cid(self):
+        return self._cid
 
-    # get column definitions and constraints for all fields
-    for field in cid._field_formats:
-        column_def, constraint = field.as_sql(dialect)
-        result += column_def + ",\n"
+    def sql_fields(self):
+        """
+        Tuples `(field_name, field_type, length, precision, is_not_null, default_value)`
+        """
+        for field in self._cid.field_formats:
+            sql_type, sql_length, sql_precision = (field.sql_ansi_type() + (None, None))[:3]
+            assert sql_type in ('varchar', 'decimal', 'int', 'date')
 
-        if len(constraint) > 0:
-            constraints += constraint + ",\n"
+            row = (field.field_name, sql_type, sql_length, sql_precision, field.is_allowed_to_be_empty,
+                   field.empty_value)
+            yield row
 
-    constraints = constraints.rsplit(',', 1)[0]
+    def create_table_statement(self):
+        result = "create table " + self._table + " (\n"
+        first_field = True
 
-    result += constraints
+        # get column definitions for all fields
+        for field in self.sql_fields():
```

Comment:
Use
```python
for field_name, field_type, length, precision, is_not_null, default_value in self.sql_fields(): ...
```
to avoid difficult to read `field[0]` etc.